driver-core/src/test/functional/com/mongodb/client/model/mql/TypeMqlValuesFunctionalTest.java

```diff
@@ -189,14 +187,25 @@ public void asStringTest() {
 
         // this is equivalent to $dateToString
         assertExpression("1970-01-01T00:00:00.123Z", of(Instant.ofEpochMilli(123)).asString());
+    }
+
+    @Test
+    public void asStringTestNestedPre82() {
+        assumeTrue(serverVersionLessThan(8, 2));
 
         // Arrays and documents are not (yet) supported:
         assertThrows(MongoCommandException.class, () ->
-                assertExpression("[]", ofIntegerArray(1, 2).asString()));
-        assertThrows(MongoCommandException.class, () ->
-                assertExpression("[1, 2]", ofIntegerArray(1, 2).asString()));
+                assertExpression("[1,2]", ofIntegerArray(1, 2).asString()));
         assertThrows(MongoCommandException.class, () ->
-                assertExpression("{a: 1}", of(Document.parse("{a: 1}")).asString()));
+                assertExpression("{\"a\":1}", of(Document.parse("{a: 1}")).asString()));
+    }
+
+    @Test
+    public void asStringTestNested() {
+        assumeTrue(serverVersionAtLeast(8, 2));
```

Comment:
Is this running against [latest](https://github.com/mongodb/mongo-java-driver/blob/main/.evergreen/.evg.yml#L1734) axis?